.github/workflows/update-changelog.yaml

```diff
@@ -0,0 +1,57 @@
+name: '[Release] Update CHANGELOG.md'
+on:
+  release:
+    types: [released]
```

Comment:
Will this run when we generate the pre-release or only the final release?

Reply:
Only after the final release. The `prereleased` activity type has some caveats: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

> he prereleased type will not trigger for pre-releases published from draft releases, but the published type will trigger. If you want a workflow to run when stable and pre-releases publish, subscribe to published instead of released and prereleased.

So I just used released to keep it simple.